### DISCUSSION TITLE
Outside collaborator best practices in GitHub Enterprise

### DISCUSSION THREAD

**Discussion by AndrWeisR** [community/community#78715] [Enterprise]
Hi all, I am looking for best practices for managing Outside Collaborators who will have write access to only selected repos, in an Enterprise Organisation.

I can either add them as organisation members and then add them to a dedicated team, but if I don't want that team to have visibility of other repos internal to the organisation, then that requires me to make all other repos secret and explicitly grant read access to other teams, which is not optimal (I'd rather have all internal repos visible to the entire organisation).

Or I can invite people as Outside Collaborators, but then I can't organise them into teams and have to invite people one by one to each repository that they need access to. Also not optimal.

So in your opinion, what's the best practice for inviting a team of outside collaborators to some of your repos.

**Comment by queenofcorgis** [comment]
Hi there @AndrWeisR 👋🏼 

Thanks for posting and I certainly hope others weigh in, but wanted to share our[ blog post](https://github.blog/2023-08-02-best-practices-for-organizations-and-teams-using-github-enterprise-cloud/) which covers best practices around organizing organizations and teams. 

Though, if you have Enterprise Managed Users (EMUs), the [Guest Collaborators](https://github.blog/changelog/2023-06-21-emu-innersource-restricted-users-private-beta-is-now-guest-collaborators/) might be a better option.

**Reply by vicker** [reply]
Hi, @queenofcorgis .

May I ask if the Guest Collaborators feature is still not public available yet?

**Reply by queenofcorgis** [reply]
@vicker it is is currently in Public Beta. Keep an eye on our [roadmap](https://github.com/orgs/github/projects/4247/views/1?filterQuery=is%3Aopen+-status%3A%22Q4+2023+%E2%80%93+Oct-Dec%22%2C%22Q2+2024+%E2%80%93+Apr-Jun%22+) and [changelog](https://github.blog/changelog/) for the latest updates.

**Comment by AndrWeisR** [comment]
Thanks @queenofcorgis  . I'm not using an IdP, so it looks like guest collaborators are not suitable. 

I've decided to have separate orgs within my enterprise - one which has external vendors as org members, with access only to a select few repos, and another org for internal repos that external vendors have no access to. That way, my internal repos can remain "internal", and hence visible to the entire internal org, and the teams can be visible to the entire org. 

By contrast, the external vendor org has private repos and secret teams, so team and repo visibility can be tightly controlled within that org.

**Reply by queenofcorgis** [reply]
@AndrWeisR appreciate you writing back and sharing how you're setting up for outside collaborators!
